private/syntax/compile/binding-spec.rkt

```diff
@@ -30,18 +30,23 @@
 
   (define bspec-elaborated (elaborate-bspec bspec-stx))
   (check-affine-pvar-use! bspec-elaborated)
+  (check-ellipsis-depth bspec-elaborated)
   (define bspec-with-implicits (add-implicit-pvar-refs bspec-elaborated bound-pvars))
   (define bspec-flattened (bspec-flatten-groups bspec-with-implicits))
+  ; might need to re-flatten after distributing
+  (define bspec-distributed-ellipses (bspec-flatten-groups (bspec-distribute-ellipses bspec-flattened)))
+  (define bspec-absorbed-ellipses (bspec-absorb-ellipses-into-imports bspec-distributed-ellipses))
```

Comment:
I don't like this and I'm not sure if there is a case that can break this. But I couldn't come up with a better idea